control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go

```diff
@@ -82,6 +82,10 @@ var (
 		"0000_50_cloud-credential-operator_01-operator-config.yaml",
 		"0000_50_cluster-authentication-operator_02_config.cr.yaml",
 		"0000_90_etcd-operator_03_prometheusrule.yaml",
+
+		// TODO: Remove when cluster-csi-snapshot-controller-operator stops shipping
+		// its ibm-cloud-managed deployment.
+		"0000_50_cluster-csi-snapshot-controller-operator_07_deployment-ibm-cloud-managed.yaml",
```

Comment:
You should not only remove the deployment from the payload but delete it if it exists on the guest cluster (upgrade case). See https://github.com/openshift/hypershift/blob/main/control-plane-operator/controllers/hostedcontrolplane/cvo/reconcile.go#L208-L247

Reply:
I thought we don't plan to support upgrade from older HyperShift ;-).
But yeah, I added the deployment there.